nvflare/app_common/widgets/streaming.py

```diff
@@ -297,6 +298,8 @@ def __init__(self, log_level=""):
 
     def handle_event(self, event_type: str, fl_ctx: FLContext):
         if event_type == EventType.ABOUT_TO_START_RUN:
+            # initialize the LocalLogger to keep all the local handlers.
+            LocalLogger.initialize()
```

Comment:
LocalLogger.initialize() should be called from the framework before firing the ABOUT_TO_START_RUN event, to be safe. Otherwise someone could write another widget that adds a handler before this streaming widget's ABORT_TO_START_RUN is called.

Reply:
Done